src/v/raft/replicate_entries_stm.cc

```diff
@@ -230,7 +230,8 @@ replicate_entries_stm::apply(std::vector<ss::semaphore_units<>> u) {
           });
           // Wait until all RPCs will be dispatched
           return _dispatch_sem.wait(requests_count)
-            .then([append_result, u = std::move(u)]() mutable {
+            .then([this, append_result, u = std::move(u)]() mutable {
+                _req.reset();
```

Comment:
do you think we should pass in the append entries request to `apply` by value instead of making it a member of the stm class? then we don't need to reset or use a unique pointer and the lifetime will be obvious from the control flow?